src/EF2680.c

```diff
@@ -3,44 +3,443 @@
+            enemy->territory->wander.wanderSizeX, enemy->territory->wander.wanderSizeZ)) {
+        npc->yaw = atan2(npc->pos.x, npc->pos.z, enemy->territory->wander.point.x, enemy->territory->wander.point.z);
+    } else {
+        npc->yaw = clamp_angle((npc->yaw + rand_int(60)) - 30.0f);
+    }
+    npc->currentAnim.w = enemy->animList[1];
+    script->functionTemp[1] = 0;
+    if (enemy->territory->wander.moveSpeedOverride < 0) {
+        npc->moveSpeed = aiSettings->moveSpeed;
+    } else {
+        npc->moveSpeed = enemy->territory->wander.moveSpeedOverride / 32767.0;
+    }
+    enemy->varTable[4] = npc->pos.y * 100.0;
+    script->functionTemp[0] = 1;
+}
 
 INCLUDE_ASM(s32, "EF2680", func_80240220_EF2830);
 
 #include "world/common/DeadUnkNpcAIFunc1.inc.c"
 
-INCLUDE_ASM(s32, "EF2680", func_80240908_EF2F18);
+#include "world/common/UnkFunc4.inc.c"
+
+#include "world/common/UnkNpcAIFunc2.inc.c"
+
+// Copy from flo_25 (CF2C10.c)
+extern f32 D_802444BC_EF6ACC[];
 
-INCLUDE_ASM(s32, "EF2680", func_80240C4C_EF325C);
+void func_80240CC4_EF32D4(Evt* script, NpcAISettings* aiSettings, EnemyTerritoryThing* territory) {
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
 
-INCLUDE_ASM(s32, "EF2680", func_80240CC4_EF32D4);
+    npc->pos.y += D_802444BC_EF6ACC[npc->duration++];
+    if (npc->duration >= 5) {
+        script->functionTemp[0] = 12;
+    }
+}
 
 #include "world/common/DeadUnkNpcAIFunc14.inc.c"
 
-INCLUDE_ASM(s32, "EF2680", func_80240E64_EF3474);
+#include "world/common/UnkNpcAIFunc3.inc.c"
 
-INCLUDE_ASM(s32, "EF2680", func_80240EC8_EF34D8);
+#include "world/common/UnkFunc6.inc.c"
 
-INCLUDE_ASM(s32, "EF2680", func_80241258_EF3868);
+#include "world/common/UnkFunc5.inc.c"
 
 INCLUDE_ASM(s32, "EF2680", func_8024137C_EF398C);
 
-INCLUDE_ASM(s32, "EF2680", func_80241594_EF3BA4);
+s32 func_80241594_EF3BA4(void) {
+    s32 i;
+
+    for (i = 50; i < 54; i++) {
+        if (get_enemy(i)->varTable[10] == 0) {
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+void func_802415DC_EF3BEC(Evt* script, NpcAISettings* aiSettings, EnemyTerritoryThing* territory) {
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
+    f32 x, y, z, w;
+    s32 var;
+    f32 temp_f20;
+    f32 temp_f22;
+    f32 temp_f24;
+    f32 var1 = enemy->varTable[4];
+    f32 var2 = enemy->varTable[3];
+    f32 var3;
+
+    temp_f22 = var1 / 100.0;
+    temp_f24 = var2 / 100.0;
+    if (npc->duration > 0) {
+        npc->duration--;
+    }
+    var3 = enemy->varTable[1];
+    temp_f20 = var3 / 100.0;
+
+    x = npc->pos.x;
+    y = npc->pos.y;
+    z = npc->pos.z;
+    w = 1000.0f;
+
+    npc_raycast_down_sides(npc->unk_80, &x, &y, &z, &w);
+    npc->pos.y = y + temp_f24 + (sin_deg(enemy->varTable[2]) * temp_f20);
+    enemy->varTable[2] = clamp_angle(enemy->varTable[2] + 0xC);
+
+    if (aiSettings->unk_14 >= 0) {
+        if (script->functionTemp[1] <= 0) {
```

Comment:
Can this if be &&d with the previous?

Reply:
Can't be done, because of `script->functionTemp[1]--;` on line 118, which only requires the first statement to be true, but not the second one.

Reply:
ahh kk